### PR TITLE
Some smaller cleanups (mostly regarding calls)

### PR DESCRIPTION
**Pull request by axic**
Depends on #188.

### CODE REVIEW

File: lib/opFns.js [hunk 1]
```diff
@@ -616,18 +617,22 @@ module.exports = {
...
-    }
+      } else {
+        stateManager.getContractCode(toAddress, function (err, code, compiled) {
+          if (err) return done(err)
+          options.compiled = compiled || false
+          options.code = code
+          makeCall(runState, options, localOpts, done)
+        })
+      }
+    })
```
Comment (holgerd77) [inline lib/opFns.js]: Hmm, code looks correct, not completely understand the changing semantics here, so probably for someone else to review again.

File: lib/opFns.js [hunk 2]
```diff
@@ -670,8 +675,8 @@ module.exports = {
       } else {
         stateManager.getContractCode(toAddress, function (err, code, compiled) {
           if (err) return done(err)
+          options.compiled = compiled || false
```
Comment (holgerd77) [inline lib/opFns.js]: Ok.

File: lib/opFns.js [hunk 3]
```diff
@@ -921,7 +930,7 @@ function makeCall (runState, callOptions, localOpts, cb) {
         runState.contract.nonce = new BN(runState.contract.nonce).subn(1)
       }
 
-      cb(err, Buffer.from([results.vm.exception]))
+      cb(null, Buffer.from([results.vm.exception]))
```
Comment (holgerd77) [inline lib/opFns.js]: All above ok.

File: lib/runCode.js [hunk 4]
```diff
@@ -173,14 +173,17 @@ module.exports = function (opts, cb) {
...
           // save result to the stack
           if (result) {
             // NOTE: Ensure that every stack item is padded to 256 bits.
             //       This should be done at every opcode in the future.
             result = utils.setLengthLeft(result, 32)
             runState.stack.push(result)
           }
-          cb(err)
+
+          cb()
```
Comment (holgerd77) [inline lib/runCode.js]: Call with ``null`` here?

File: lib/opFns.js (hunk 1 above)
Comment (axic) [inline lib/opFns.js]: The account must exists even for precompiles.

File: lib/runCode.js (hunk 4 above)
Comment (axic) [inline lib/runCode.js]: Good catch, though that file has all the callbacks without null, need to tackle it at some point. Also some other methods (`parseVmResults`) there have an issue passing both `err` and `results`. Haven't reviewed it yet.

**Pull request merged**